include/bout/array.hxx

```diff
@@ -161,17 +161,19 @@ public:
   /*!
    * Returns true if the Array is empty
    */
-  bool empty() const {
+  bool empty() const noexcept {
     return ptr == nullptr;
   }
 
   /*!
    * Return size of the array. Zero if the array is empty.
    */
-  int size() const {
+  int size() const noexcept {
     if(!ptr)
       return 0;
 #ifdef BOUT_ARRAY_WITH_VALARRAY
+    // Note: std::valarray::size is not noexcept, so Array::size
+    // shouldn't be either if we're using valarrays
```

Comment:
But we're not doing that here? Should we wrap the signature in the `ifdef` as well to allow this?